Lib/multiprocessing/spawn.py

```diff
@@ -43,7 +43,10 @@ def set_executable(exe):
     _python_exe = exe
 
 def get_executable():
-    return _python_exe
+    if sys.platform == 'win32':
+        return os.fsdecode(_python_exe)
+    else:
+        return os.fsencode(_python_exe)
```

Comment:
I'd prefer to do the conversion once when setting the value. How about this?

```python
if sys.platform != 'win32':
    WINEXE = False
    WINSERVICE = False
else:
    WINEXE = getattr(sys, 'frozen', False)
    WINSERVICE = sys.executable.lower().endswith("pythonservice.exe")

# Embedding applications need to set the path of the Python executable.

def set_executable(exe):
    global _python_exe
    if sys.platform == 'win32':
        _python_exe = os.fsdecode(exe)
    else:
        _python_exe = os.fsencode(exe)

def get_executable():
    return _python_exe

if WINSERVICE:
    set_executable(os.path.join(sys.exec_prefix, 'python.exe'))
else:
    set_executable(sys.executable)

```

Reply:
I agree, since this is the only entry point for a user-provided path (besides rebinding `spawn._python_exe`, but this is not part of the API), I think it makes sense to avoid paying the conversion cost at each invocation of `spawn.get_executable()`. Updated.